conductr_cli/conduct_backup.py

```
@@ -0,0 +1,262 @@
+                                   headers={"Accept": "multipart/form-data"})
+
+    result = []
+    # ensure that the bundle hasn't been unloaded
+    if requests.codes.not_found != response.status_code:
+        validation.raise_for_status_inc_3xx(response)
+        multipart_data = MultipartDecoder.from_response(response)
+        for part in multipart_data.parts:
+            result.append(part.content)
+    else:
+        log.error("Bundle unavailable")
+        raise ConductBackUpError("Bundle was not found.")
+
+    return result
+
+
+@validation.handle_connection_error
+@validation.handle_http_error
+def members(args):
+    log = logging.getLogger(__name__)
+    url = conduct_url.url('members', args)
+    response = conduct_request.get(args.dcos_mode, conductr_host(args), url, auth=args.conductr_auth,
+                                   verify=args.server_verification_file, timeout=DEFAULT_HTTP_TIMEOUT)
+    validation.raise_for_status_inc_3xx(response)
+    log.debug(validation.pretty_json(response.text))
+    return response.text
+
+
+@validation.handle_connection_error
+@validation.handle_http_error
+def agents(args):
+    log = logging.getLogger(__name__)
+    url = conduct_url.url('agents', args)
+    response = conduct_request.get(args.dcos_mode, conductr_host(args), url, auth=args.conductr_auth,
+                                   verify=args.server_verification_file, timeout=DEFAULT_HTTP_TIMEOUT)
+    validation.raise_for_status_inc_3xx(response)
+    log.debug(validation.pretty_json(response.text))
+    return response.text
+
+
+def create_backup_directory(output_path):
+    backup_directory_format = "%Y-%m-%d-%H-%M"
+    backup_name = str(time.strftime(backup_directory_format, time.gmtime(time.time())))
+
+    backup_path = os.path.join(output_path, backup_name)
+    if not os.path.exists(backup_path):
+        try:
+            os.makedirs(backup_path, exist_ok=True)
+        except OSError as e:
+            raise ConductBackUpError(e)
```

I think it's better to let the original error propagate - just call `os.makedirs` and let it fail if it fails.

---

done